libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts

```diff
@@ -5010,6 +5056,52 @@ describe('Lookup component', function () {
               closeModal(fixture);
             }));
 
+            it('should add items when scrolling ends with "Only show selected" active', async () => {
+              component.enableShowMore = true;
+              component.selectedFriends = [
+                ...component.data.filter((item) => !item.description),
+              ];
+
+              fixture.detectChanges();
+              await fixture.whenStable();
+
+              triggerInputFocus(fixture);
+              fixture.detectChanges();
+              await fixture.whenStable();
+              // Not using `clickShowMore` due to it being for `fakeAsync`
+              SkyAppTestUtility.fireDomEvent(getShowMoreButton(), 'mousedown');
+              fixture.detectChanges();
+              await fixture.whenStable();
+              fixture.detectChanges();
+
+              // Not using `selectOnlyShowSelected` due to it being for `fakeAsync`
+              (
+                document.querySelector(
+                  '.sky-lookup-show-more-modal-muiltiselect-toolbar .sky-toolbar-view-actions input'
+                ) as HTMLElement
+              ).click();
+              fixture.detectChanges();
+              await fixture.whenStable();
+              fixture.detectChanges();
+
+              expect(getRepeaterItemCount()).toBe(10);
+
+              const modalContent = document.querySelector('.sky-modal-content');
+              modalContent.scrollTop = modalContent.scrollHeight;
+              SkyAppTestUtility.fireDomEvent(modalContent, 'scroll');
+              fixture.detectChanges();
+              await fixture.whenStable();
+              fixture.detectChanges();
+
+              expect(getRepeaterItemCount()).toBe(18);
+
+              (
+                document.querySelector(
+                  '.sky-lookup-show-more-modal-close'
+                ) as HTMLElement
+              )?.click();
```

Comment:
Can any of this duplicate code be placed in a helper function so if we need to make changes to both tests the logic lives in one place?

Reply:
Added more helper methods